src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -182,8 +182,9 @@ public SCM build(SCMHead head, @CheckForNull SCMRevision revision) {
                 descriptor.getBitbucketScmHelper(serverConfiguration.getBaseUrl(),
                         globalCredentialsProvider.getGlobalAdminCredentials().orElse(null));        
         
-        BitbucketDefaultBranch defaultBranch = scmHelper.getDefaultBranch(repository.getProjectName(), repository.getRepositoryName());
-        return Collections.singletonList(new BitbucketRepositoryMetadataAction(repository, defaultBranch));
+        scmHelper.getDefaultBranch(repository.getProjectName(), repository.getRepositoryName())
+                .ifPresent(defaultBranch -> result.add(new BitbucketRepositoryMetadataAction(repository, defaultBranch)));
+        return result;
     }
 
     @NonNull
```

Comment:
we don't need `@NonNull` as the package level annotations are already indicating this. (could be more than this place)